lib/ansiblelint/utils.py

```diff
@@ -181,19 +182,24 @@ def find_children(playbook, playbook_dir):
     basedir = os.path.dirname(playbook[0])
     items = _playbook_items(playbook_ds)
     for item in items:
-        for child in play_children(basedir, item, playbook[1], playbook_dir):
-            if "$" in child['path'] or "{{" in child['path']:
-                continue
-            valid_tokens = list()
-            for token in split_args(child['path']):
-                if '=' in token:
-                    break
-                valid_tokens.append(token)
-            path = ' '.join(valid_tokens)
-            results.append({
-                'path': path_dwim(basedir, path),
-                'type': child['type']
-            })
+        try:
+            for child in play_children(basedir, item, playbook[1], playbook_dir):
```

Comment:
Here's how you could solve this:
```suggestion
            def rebind_match_filename(filename, func):
                def func_wrapper(*args, **kwargs):
                    try:
                        return func(*args, **kwargs)
                    except MatchError as e:
                        e.filename = filename
                        raise e
                return func_wrapper
            for child in rebind_match_filename(playbook[0], play_children)(basedir, item, playbook[1], playbook_dir):
```